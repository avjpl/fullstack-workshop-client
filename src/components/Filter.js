import React, { Fragment } from 'react';

const Filter = ({ onFilterChange, selected }) => (
  <Fragment>
    <p style={styles.show}>Show:</p>
    <select
      value={selected}
      style={styles.select}
      onChange={({ target: { value } }) => {
        onFilterChange(value);
      }}
    >
      <option value="POPULARITY">Popular Movies</option>
      <option value="RELEASE_DATE">New Movies</option>
      <option value="LIKES">Liked Movies</option>
    </select>
  </Fragment>
);

const styles = {
  show: {
    display: 'inline',
  },
  select: {
    marginBottom: 9,
  },
};

export default Filter;
