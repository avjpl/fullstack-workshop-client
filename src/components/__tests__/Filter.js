import React from 'react';
import Filter from '../Filter';
import renderer from 'react-test-renderer';

describe('Filter', () => {
  it('renders', () => {
    const comp = renderer.create(<Filter />);
    expect(comp.toJSON()).toMatchSnapshot();
  });

  it('allows control from `selected` prop', () => {
    const comp = renderer.create(<Filter selected="LIKES" />);
    const select = comp.root.findByType('select');
    expect(select.props.value).toEqual('LIKES');
  });
});
