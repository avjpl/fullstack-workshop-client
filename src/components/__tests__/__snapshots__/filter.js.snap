// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter renders 1`] = `
Array [
  <p
    style={
        Object {
            "display": "inline",
          }
    }
>
    Show:
</p>,
  <select
    onChange={[Function]}
    style={
        Object {
            "marginBottom": 9,
          }
    }
    value={undefined}
>
    <option
        value="POPULARITY"
    >
        Popular Movies
    </option>
    <option
        value="RELEASE_DATE"
    >
        New Movies
    </option>
    <option
        value="LIKES"
    >
        Liked Movies
    </option>
</select>,
]
`;
